core/src/main/java/org/jruby/java/addons/ClassJavaAddons.java

```diff
@@ -65,7 +66,13 @@ private static IRubyObject becomeJava(final ThreadContext context, final RubyCla
 
         klass.reifyWithAncestors(dumpDir, useChildLoader);
 
-        final Class<?> reifiedClass = klass.getReifiedClass();
+        Class<?> reifiedClass = klass.getReifiedClass();
+        if (reifiedClass == null) { // java proxies can't be reified, but they deserve field accessors too
+            reifiedClass = JavaProxyClass.getProxyClass(context.getRuntime(), klass).getJavaClass();
+        }
+        if (reifiedClass == null) {
+            throw context.runtime.newTypeError("requested class " + klass.getName() + " was not reifiable");
+        }
```

Comment:
@headius This bit of code does not distinguish between Failed Reified Ruby classes and Concretized Classes. getReified is null if either it's a failure to reify or a java proxy class. Should the logic be split between those two cases, and if so, how?

Reply:
Aha, so you think maybe we need another way to know a given class is a java proxy class. Perhaps we add another boolean flag to RubyClass like `isJavaSubclass` or `isJavaProxy`?

Reply:
>  so you think maybe we need...
No, I am asking if we do. I have no strong opinion on these internals, but I do think the error message could be a misleading if we care to tell users about the difference between concrete vs reified, so I'm wondering if *you* think we should disambiguate. 
Secondarily, I think it's safe for this code path to be retried for failed reified and failed concrete multiple times, but that might have implications I'm not aware of